### PR TITLE
ci: added github tooling

### PR DESCRIPTION
**Pull request by shanduur**
## What this PR does / why we need it:

Initial CI configuration. Cherry-picked from #1 .

## Which issue(s) this PR resolves:

N/A

## Special notes for your reviewer:

Tests will fail, as there is no code to run them against. After #2 and #3 are merged, the rebase should make them execute succesfully.

## Additional documentation e.g., enhancement proposals, usage docs, etc.:

N/A

### CODE REVIEW

File: .github/workflows/10-linters-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache: false
+      - uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          skip-cache: true
+
+  shell-linter:
+    name: Run Shellcheck, Checkmake and Perl Critic
```
Comment (avestuk) [inline .github/workflows/10-linters-tests.yaml:42]: Perl Critic?

**Review by avestuk — APPROVED** [review]

File: .github/workflows/10-linters-tests.yaml (hunk 1 above)
Comment (shanduur) [reply .github/workflows/10-linters-tests.yaml:42]: Whoopsie, gonna clean this name, I copied it from my template 😮‍💨

File: .github/workflows/00-labels-sync.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    name: Sync labels
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-go@v4
+      with:
+        go-version: '1.20'
+    - run: go install github.com/shanduur/labeler/cmd/labeler@main
```
Comment (luthermonson) [inline .github/workflows/00-labels-sync.yaml:30]: if youre gonna keep using this thing like this you should make an action for it

File: .github/workflows/99-release.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+env:
+  REGISTRY: docker.io
+
+jobs:
+  docker:
+    name: Build dev image and run scans
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up QEMU
```
Comment (luthermonson) [inline .github/workflows/99-release.yaml:21]: i dont think we need the qemu and buildx stuff just yet

**Review by luthermonson — APPROVED** [review]

**Review by avestuk — APPROVED** [review]

**Pull request merged**